bindings/pyroot/pythonizations/src/RooFitPyz.cxx

```diff
@@ -0,0 +1,64 @@
+// Author: Jonas Rembser, CERN 07/2021
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+// Bindings
+#include "CPyCppyy/API.h"
+#include "PyzCppHelpers.hxx"
+
+// RooFit
+#include "RooFitPyzHelpers.h"
+
+// Standard library
+#include <stdexcept>
+
+namespace RooFit {
+namespace Detail {
+
+// Helper function to create the intermediate vector for the RooArgSet constructor from a Python set.
+std::vector<RooAbsArgPtrOrDouble> rooAbsArgPtrOrDoubleFromPyObject(PyObject *pyobject)
+{
+   // check that the Python object is a tuple
+   if (!PySet_Check(pyobject)) {
+      throw std::runtime_error(
+         "RooArgSet creation from a Python object only works with Python lists, tuples, and sets.");
+   }
+
+   // prepare the output vector
+   std::vector<RooAbsArgPtrOrDouble> argVector;
+   if (PySet_Size(pyobject) == 0)
+      return argVector;
+   argVector.reserve(PySet_Size(pyobject));
+
+   // adapted from the template given in the documentation of the iterator protocol in the Python C API:
+   // https://docs.python.org/3/c-api/iter.html
+   PyObject *iterator = PyObject_GetIter(pyobject);
+   PyObject *item;
+   void *ptr;
+   while ((item = PyIter_Next(iterator))) {
+
+      if (PyFloat_Check(item)) {
+         // if the item is a Python float, add the value to the output vector
+         argVector.push_back(PyFloat_AsDouble(item));
+      } else if ((ptr = CPyCppyy::Instance_AsVoidPtr(item)) &&
```

Comment:
To check that this is a CPPInstance, you can use `CPPInstance_Check` defined in `CPPInstance.h`, see e.g.:

https://github.com/root-project/root/blob/master/bindings/tpython/src/TPyReturn.cxx#L182